crates/ruff_linter/src/rules/flake8_quotes/rules/avoidable_escaped_quote.rs

```diff
@@ -48,21 +49,65 @@ impl AlwaysFixableViolation for AvoidableEscapedQuote {
     }
 }
 
+/// ## What it does
+/// Checks for strings that include escaped quotes not matching the outer
+/// quotes, and suggests removing the unnecessary backslash.
+///
+/// ## Why is this bad?
+/// It's escaping something that does not need escaping.
+///
+/// ## Example
+/// ```python
+/// foo = "bar\'s"
+/// ```
+///
+/// Use instead:
+/// ```python
+/// foo = "bar's"
+/// ```
+///
+/// ## Formatter compatibility
+/// We recommend against using this rule alongside the [formatter]. The
+/// formatter automatically removes unnecessary escapes, making the rule
+/// redundant.
+///
+/// [formatter]: https://docs.astral.sh/ruff/formatter
+#[violation]
+pub struct UnnecessaryEscapedQuote;
+
+impl AlwaysFixableViolation for UnnecessaryEscapedQuote {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Do not escape quotes that do not need escaping")
```

Comment:
If you can some up with a better text please change it. As usual, naming is hard...